cpp/src/neighbors/ivf_pq/ivf_pq_build.cuh

```diff
@@ -1754,7 +1796,13 @@ auto build(raft::resources const& handle,
       cluster_centers, index.n_lists(), index.dim());
     cuvs::cluster::kmeans::balanced_params kmeans_params;
     kmeans_params.n_iters = params.kmeans_n_iters;
-    kmeans_params.metric  = static_cast<cuvs::distance::DistanceType>((int)index.metric());
+    if (index.metric() == distance::DistanceType::CosineExpanded) {
```

Comment:
`cuvs::cluster::kmeans_balanced::fit` Already supports Cosine metric, so there is no need to add normalization + switch to inner product.